From: Andrea Arcangeli <andrea@cpushare.com>
Subject: [PATCH seccomp: make tsc disabling optional
Patch-mainline: unknown
References: 191123

Make the TSC disable purely paranoid feature optional, so by default seccomp
returns absolutely zerocost.

Ported from 2.6.19 to 2.6.24-rc7 by Jeff Mahoney.
Addition of x86-64 by Jan Beulich.

Signed-off-by: Andrea Arcangeli <andrea@cpushare.com>
Acked-by: Jeff Mahoney <jeffm@suse.com>
Automatically created from "patches.fixes/seccomp-disable-tsc-option" by xen-port-patches.py

--- head-2009-07-28.orig/arch/x86/kernel/process-xen.c	2009-07-28 14:50:00.000000000 +0200
+++ head-2009-07-28/arch/x86/kernel/process-xen.c	2009-07-29 10:17:08.000000000 +0200
@@ -132,6 +132,7 @@ static void hard_disable_TSC(void)
 
 void disable_TSC(void)
 {
+#ifdef CONFIG_SECCOMP_DISABLE_TSC
 	preempt_disable();
 	if (!test_and_set_thread_flag(TIF_NOTSC))
 		/*
@@ -140,6 +141,7 @@ void disable_TSC(void)
 		 */
 		hard_disable_TSC();
 	preempt_enable();
+#endif
 }
 
 static void hard_enable_TSC(void)
